pandas/tests/frame/methods/test_combine_first.py

```diff
@@ -217,10 +219,13 @@ def test_combine_first_timezone(self):
             },
             columns=["UTCdatetime", "abc"],
             index=pd.date_range("20140627", periods=2, freq="D"),
+            dtype="object",
         )
-        tm.assert_frame_equal(res, exp)
         assert res["UTCdatetime"].dtype == "datetime64[ns, UTC]"
         assert res["abc"].dtype == "datetime64[ns, UTC]"
+        # GH Issue 7509
```

Comment:
can you split this to a separate test (the added code and what is needed to repro)

Reply:
Is not possible to remove all added `dtype="object"` ? The `df1` and `df2` now changed to no longer have object dtype due to the fix, but this actually seems a more correct case to test. Or does `combine_first` not handle that correctly / still return object?

Reply:
Fixing the setitem bug made these datetime64ns rather than object. But then `combine_first` is buggy, as described in https://github.com/pandas-dev/pandas/issues/7509.

So yes, ideally we test non-object, but https://github.com/pandas-dev/pandas/issues/7509 will have to be fixed first (and shouldn't block this PR).

Reply:
Ah, thanks for the pointer. Yes certainly shouldn't block the PR. 
It might be worth adding a comment explaining this here though (instead of just the github issue reference, as you need to look that up to know what it is about)

Reply:
Thanks for the recommendation, updated the comment